### PR TITLE
Tl2023 355

### PR DESCRIPTION
**Pull request by Skumarhpa**

### CODE REVIEW

File: src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs [hunk 1]
```diff
@@ -60,5 +66,84 @@ public Task<AdminAddCoreRommOutcomeViewModel> GetAdminAddPathwayRommOutcomeAsync
...
+        private async Task<List<LookupViewModel>> GetAdminAddResultGrades(LookupCategory lookupCategory)
+        {
+            IList<LookupData> grades = await _internalApiClient.GetLookupDataAsync(lookupCategory);
+            return _mapper.Map<List<LookupViewModel>>(grades);
+        }
+
+        private async Task<List<LookupViewModel>> GetAdminChangeResultGrades(LookupCategory lookupCategory, string grade, bool isRomm = false)
+        {
+            IList<LookupData> grades = await _internalApiClient.GetLookupDataAsync(lookupCategory);
+            grades.Remove(grades.Where(t => t.Value == grade).FirstOrDefault());
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs:118]: grades.Remove(grades.FirstOrDefault(t => t.Value == grade))

File: src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs [hunk 2]
```diff
@@ -60,5 +66,84 @@ public Task<AdminAddCoreRommOutcomeViewModel> GetAdminAddPathwayRommOutcomeAsync
...
+            if (!isRomm) grades.Insert(grades.Count, new LookupData { Code = Constants.NotReceived, Value = Content.Result.ManageSpecialismResult.Option_Remove_Result });
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs:119]: Why grade.Insert and no grade.Add? You're adding an item at the end of the list right?

File: src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs [hunk 3]
```diff
@@ -60,5 +66,84 @@ public Task<AdminAddCoreRommOutcomeViewModel> GetAdminAddPathwayRommOutcomeAsync
...
+
+
+        private List<LookupViewModel> GetAdminAddRommOutcomeChangeGradeSpecialismGrades(List<LookupViewModel> Grades)
+        {
+            return Grades.Where(t => !t.Code.Equals(Constants.SpecialismComponentGradeQpendingResultCode, StringComparison.InvariantCultureIgnoreCase)
+            && !t.Code.Equals(Constants.SpecialismComponentGradeXNoResultCode, StringComparison.InvariantCultureIgnoreCase)
+            && !t.Code.Equals(Constants.NotReceived, StringComparison.InvariantCultureIgnoreCase)).ToList();
+        }
+
+        private async Task<List<LookupViewModel>> GetAdminAddResultGrades(LookupCategory lookupCategory)
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs:109]: This method is not used at all

File: src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs [hunk 4]
```diff
@@ -60,5 +66,84 @@ public Task<AdminAddCoreRommOutcomeViewModel> GetAdminAddPathwayRommOutcomeAsync
...
+            Task<AdminLearnerRecord> learnerRecordTask = _internalApiClient.GetAdminLearnerRecordAsync(registrationPathwayId);
+            Task<IList<LookupData>> gradesTask = _internalApiClient.GetLookupDataAsync(lookupCategory);
+
+            await Task.WhenAll(learnerRecordTask, gradesTask);
+
+            AdminLearnerRecord learnerRecord = learnerRecordTask.Result;
+            IList<LookupData> grades = gradesTask.Result;
+            if (ischange)
+            {
+                grades.Insert(grades.Count, new LookupData { Code = Constants.NotReceived, Value = Content.Result.ManageSpecialismResult.Option_Remove_Result });
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs:135]: Add?

File: src/Sfa.Tl.ResultsAndCertification.Web/Properties/launchSettings.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
     "anonymousAuthentication": true,
     "iisExpress": {
       "applicationUrl": "http://localhost:51975",
-      "sslPort": 44345
+      "sslPort": 44320
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Properties/launchSettings.json:7]: Do we need to check this file in?

Comment (Skumarhpa) [reply src/Sfa.Tl.ResultsAndCertification.Web/Properties/launchSettings.json:7]: This is to revert to original values.

**Review by ibai-gonzalez — APPROVED** [review]

**Pull request merged**